src/libraries/System.Numerics.Tensors/tests/TensorPrimitivesTests.cs

```diff
@@ -58,30 +58,56 @@ private static unsafe float MathFMinMagnitude(float x, float y)
             return (ax < ay) || float.IsNaN(ax) || (ax == ay && *(int*)&x < 0) ? x : y;
         }
 
-        private static unsafe int SingleToInt32(float f) => *(int*)&f;
+        private static unsafe float UInt32ToSingle(uint i) => *(float*)&i;
 
-        private static unsafe float Int32ToSingle(int i) => *(float*)&i;
-
-        private static float AnotherSingleNaN = Int32ToSingle(-8388607);
-
-        /// <summary>Loads a variety of special values (e.g. NaN) into random positions in <paramref name="x"/>.</summary>
-        private static void SetSpecialValues(Span<float> x)
+        /// <summary>Gets a variety of special values (e.g. NaN).</summary>
+        private static IEnumerable<float> GetSpecialValues()
         {
             // NaN
-            x[s_random.Next(x.Length)] = float.NaN;
-            x[s_random.Next(x.Length)] = AnotherSingleNaN;
+            yield return UInt32ToSingle(0xFFC0_0000); // -qNaN / float.NaN
+            yield return UInt32ToSingle(0xFFFF_FFFF); // -qNaN / all-bits-set
+            yield return UInt32ToSingle(0x7FC0_0000); // +qNaN
+            yield return UInt32ToSingle(0xFFA0_0000); // -sNaN
+            yield return UInt32ToSingle(0x7FA0_0000); // +sNaN
 
             // +Infinity, -Infinity
-            x[s_random.Next(x.Length)] = float.PositiveInfinity;
-            x[s_random.Next(x.Length)] = float.NegativeInfinity;
+            yield return float.PositiveInfinity;
+            yield return float.NegativeInfinity;
 
             // +Zero, -Zero
-            x[s_random.Next(x.Length)] = +0.0f;
-            x[s_random.Next(x.Length)] = -0.0f;
+            yield return +0.0f;
+            yield return -0.0f;
 
-            // +Epsilon, -Epsilon
-            x[s_random.Next(x.Length)] = +float.Epsilon;
-            x[s_random.Next(x.Length)] = -float.Epsilon;
+            // Subnormals
+            yield return +float.Epsilon;
+            yield return -float.Epsilon;
+            yield return UInt32ToSingle(0x007F_FFFF);
+            yield return UInt32ToSingle(0x807F_FFFF);
+
+            // Normals
+            yield return UInt32ToSingle(0x0080_0000);
+            yield return UInt32ToSingle(0x8080_0000);
+            yield return UInt32ToSingle(0x7F7F_FFFF);
+            yield return UInt32ToSingle(0x8F7F_FFFF);
```

Comment:
This was probably a typo on my end, it should be:
```suggestion
            yield return UInt32ToSingle(0x7F7F_FFFF); // MaxValue
            yield return UInt32ToSingle(0xFF7F_FFFF); // MinValue
```